kernel-ewasm/cap9-std/src/syscalls.rs

```diff
@@ -395,20 +395,25 @@ impl Serialize for Call {
     }
 }
 
-impl Deserialize for Payload {
-    type Error = io::Error;
+impl Deserialize<u8> for Payload {
+    type Error = cap9_core::Error;
 
-    fn deserialize<R: io::Read>(reader: &mut R) -> Result<Self, Self::Error> {
+    fn deserialize<R: cap9_core::Read<u8>>(reader: &mut R) -> Result<Self, Self::Error> {
```

Comment:
This should use `#remaining`, I don't see where the length is being encoded here, so I'm not sure how other additional input is handled. 
```suggestion
    fn deserialize<R: cap9_core::Read<u8>>(reader: &mut R) -> Result<Self, Self::Error> {
           let mut payload: Vec<u8> = Vec::with_capacity(reader.remaining());
           payload.resize(reader.remaining(), 0);
           reader.read(&mut payload);
           payload
    }
```

Reply:
Ahh yep, that's what that TODO note says. I added `#remaining` as part of the refactor, which added the assumption of a sized buffer. I didn't update this to take advantage of `#remaining`.

Reply:
Updated with f4ff6e8a6f875b0ce60b7b8d98083828539ff39b.